components/plan-build/bin/hab-plan-build.sh

```diff
@@ -2046,12 +2046,21 @@ do_strip() {
 }
 
 # Default implementation for the `do_strip()` phase.
+# TODO(SM): Previous versions of the `find` utility reported x-pie-exectuable
+# as x-sharedlib. This means that while the intent was to `--strip-all` for 
+# x-executable, in reality we have been running `--strip-unneeded`. In order to 
+# be consistant with past behavior we will pass `--strip-unneeded` when stripping
+# x-pie-executable. In the future, we will need to make a decision as to the behavior 
```

Comment:
Do we have an issue filed for this future work? There are some other fix-ups that could make this code a bit more robust.

Reply:
Yes, I added #6191 to track the work